packages/eslint-config/rules/style.js

```diff
@@ -484,27 +484,27 @@ module.exports = {
     'no-restricted-syntax': ['error',
       {
         selector: 'ForStatement',
-        message: 'Go back to C you mad man.',
+        message: 'C-Style for loops are disallowed, consider using a higher-order array function like map/filter/reduce, a for-of statement, or a while loop.',
       },
       {
         selector: 'WithStatement',
-        message: 'ಠ_ಠ With statement is deprecated, ya chump.',
+        message: 'Use of the WITH statement is deprecated in ES5 and generally discouraged due to unpredictable behaviour.',
       },
       {
         selector: 'LabeledStatement',
-        message: 'Labels are like GOTOs, they make your code less readable for little benefit.',
+        message: 'Labeled statements are considered harmful as they break the control flow of the program and makes the behaviour of code unpredictable and difficult to understand.',
       },
       {
         selector: 'BinaryExpression[operator=\'in\']',
-        message: 'In operator can cause problematic behavior. Use Obj.isOwnProperty() or Reflect.has() instead.',
+        message: 'The In operator can cause problematic behavior. Use Obj.isOwnProperty() or Reflect.has() instead.',
       },
       {
         selector: 'no-restricted-syntax',
         message: 'Buddy you\'ve got problems.',
       },
       {
         selector: 'DebuggerStatement',
-        message: 'You may think you need this, but you don\'t. Set your devenv up correctly, chump.',
+        message: 'Manual debugger statements are not allowed, consider using the breakpoint system in your development environment.',
```

Comment:
I think we could remove this once since it's covered by no-debugger. Also I'm not entirely against the use of debugger in development as long as it's removed from production or staging builds. Just like `console.log()`

Reply:
maybe make that specific one a warning with the message to ensure that they are removed for the production build?

Reply:
so.. problem here is that we cannot degrade no-restricted-syntax to a warning. In any sort of webpack HMR dev environment with `eslint-loader` (like our nextjs projects), this will break the dev server until it's fixed, which is not great if we want to allow debugger in development mode.

I feel like this deserves a bigger breakdown on the pros and cons of debugger and our stance on it in our style guide, That discussion is not for this PR, though. I'll probably write something up on it in either an issue or PR that revises the `no-debugger` rule.